builder/dockerfile/internals.go

```diff
@@ -383,11 +383,9 @@ func (b *Builder) calcCopyInfo(cmdName, origPath string, allowLocalDecompression
 		hfi.SetHash("file:" + hfi.Hash())
 		return copyInfos, nil
 	}
-
 	// Must be a dir
-
 	var subfiles []string
-	b.context.Walk(origPath, func(path string, info builder.FileInfo, err error) error {
+	b.context.Walk(statPath, func(path string, info builder.FileInfo, err error) error {
```

Comment:
@tonistiigi I don't know why we're not checking for the error here. Do you see any reason not to return the error?